Document ENV injection to Functions (#15071)
kyma-project/kyma: docs/03-tutorials/00-serverless/svls-14-inject-envs.md

---

> title: Inject Environment Variables

Reviewer: ```suggestion
title: Inject environment variables
```

---

> This tutorial shows how to inject environment variables into funtion.

Reviewer: ```suggestion
This tutorial shows how to inject environment variables into Function.
```

> You can specify environment variables in the function definition iself or define references to kubernetes secrets or configmaps.

Reviewer: ```suggestion
You can specify environment variables in the Function definition, or define references to the Kubernetes Secrets or ConfigMaps.
```

## Prerequisites

Before you start, make sure you have these tools installed:

- Kyma installed on a cluster

## Steps

Follow these steps:

> 1. Prepare config map

Reviewer: ```suggestion
1. Create your ConfigMap
```

```bash
kubectl create configmap my-config --from-literal config-env="I come from config map"
```

> 2. Prepare secret

Reviewer: ```suggestion
2. Create your Secret
```

```bash
kubectl create secret generic  my-secret  --from-literal secret-env="I come from secret"
```

<div tabs name="steps" group="create-function">
  <details>
  <summary label="cli">
  Kyma CLI
  </summary>

> 3. Generate function configuration and sources:

Reviewer: ```suggestion
3. Generate the Function's configuration and sources:
```

    ```bash
    kyma init function --name my-function
    ```

> 4. Define ENVs as part of function configuration file. Modify `config.yaml` with the following:

Reviewer: ```suggestion
4. Define environment variables as part of the Function configuration file. Modify `config.yaml` with the following:
```

    ```yaml
    name: my-function
    namespace: default
    runtime: nodejs14
    source:
        sourceType: inline
    env:
      - name: env1
        value: "I come from function definition"
      - name: env2
        valueFrom:
          configMapKeyRef:
            name: my-config
            key: config-env
      - name: env3
        valueFrom:
          secretKeyRef:
            name: my-secret
            key: secret-env
    ```

> 5. Use injected ENVs in the handler file. Modify `handler.js` with the following:

Reviewer: ```suggestion
5. Use injected environment variables in the handler file. Modify `handler.js` with the following:
```
